internal/testrunner/runners/system/system.go

```diff
@@ -5,31 +5,364 @@
 	"github.com/elastic/elastic-package/internal/testrunner"
+	"github.com/elastic/elastic-package/internal/testrunner/runners/system/servicedeployer"
 )
 
+func init() {
+	testrunner.RegisterRunner(TestType, Run)
+}
+
 const (
 	// TestType defining system tests
 	TestType testrunner.TestType = "system"
 )
 
 type runner struct {
-	testFolderPath string
+	testFolder      testrunner.TestFolder
+	packageRootPath string
+	stackSettings   stackSettings
+}
+
+type stackSettings struct {
+	elasticsearch struct {
+		host     string
+		username string
+		password string
+	}
+	kibana struct {
+		host string
+	}
 }
 
 // Run runs the system tests defined under the given folder
 func Run(options testrunner.TestOptions) error {
-	r := runner{options.TestFolderPath}
+	r := runner{
+		options.TestFolder,
+		options.PackageRootPath,
+		getStackSettingsFromEnv(),
+	}
 	return r.run()
 }
 
 func (r *runner) run() error {
-	fmt.Println("system run", r.testFolderPath)
+	pkgManifest, err := packages.ReadPackageManifest(filepath.Join(r.packageRootPath, packages.PackageManifestFile))
+	if err != nil {
+		return errors.Wrap(err, "reading package manifest failed")
+	}
+
+	datasetPath, found, err := packages.FindDatasetRootForPath(r.testFolder.Path)
+	if err != nil {
+		return errors.Wrap(err, "locating dataset root failed")
+	}
+	if !found {
+		return errors.New("dataset root not found")
+	}
+
+	datasetManifest, err := packages.ReadDatasetManifest(filepath.Join(datasetPath, packages.DatasetManifestFile))
+	if err != nil {
+		return errors.Wrap(err, "reading dataset manifest failed")
+	}
+
+	// Step 1. Setup service.
+	// Step 1a. (Deferred) Tear down service.
+	logger.Info("setting up service...")
+	serviceRunner, err := servicedeployer.Factory(r.packageRootPath)
+	if err != nil {
+		return errors.Wrap(err, "could not create service runner")
+	}
+
+	tempDir, err := install.TempDir()
+	if err != nil {
+		return errors.Wrap(err, "could not get temporary folder")
+	}
+
+	ctxt := common.MapStr{}
+	ctxt.Put("service.name", r.testFolder.Package)
+	ctxt.Put("service.logs.folder.local", tempDir)
+	ctxt.Put("service.logs.folder.agent", "/tmp/service_logs/")
+
+	ctxt, err = serviceRunner.SetUp(ctxt)
+	defer func() {
+		logger.Info("tearing down service...")
+		if err := serviceRunner.TearDown(ctxt); err != nil {
+			logger.Errorf("error tearing down service: %s", err)
+		}
+	}()
+	if err != nil {
+		return errors.Wrap(err, "could not setup service")
+	}
+
+	// Step 2. Configure package (single data stream) via Ingest Manager APIs.
+	im, err := ingestmanager.NewClient(r.stackSettings.kibana.host, r.stackSettings.elasticsearch.username, r.stackSettings.elasticsearch.password)
+	if err != nil {
+		return errors.Wrap(err, "could not create ingest manager client")
+	}
+
+	logger.Info("creating test policy...")
+	p := ingestmanager.Policy{
+		Name:        fmt.Sprintf("ep-test-system-%s-%s", r.testFolder.Package, r.testFolder.Dataset),
```

Comment:
I didn't see you applied any timestamps here. I wonder if it's not better to include some. Consider a scenario in which developer stops the test runner in the middle and for some reason the cleanup never happened or failed. We can prevent some transient errors in the future.

Reply:
Yes, we could do that. Another option would be to not use timestamps so we have the same identifiers for each test run. Then we clean up resources _before_ (and after) the test run starts, in case any resources were left from transient failures from a previous run.

Either way, I agree we need to account for orphan resources due to transient failures from previous runs.

WDYT?

Reply:
> Then we clean up resources before (and after) the test run starts, in case any resources were left from transient failures from a previous run.

As long as we don't have intention to run tests in parallel, this can work pretty well.

Reply:
Actually, what I suggested above is not trivial to implement for policies as their IDs are generated by Kibana. We could retrieve all policies and delete ones that match a certain naming convention, but let's worry about that in a follow up PR.

For now, I took your suggestion and added a timestamp to the policy name (via 9fde2df).